src/com/github/kornilova203/matlab/resolveUtil.kt

```diff
@@ -33,25 +33,28 @@ fun PsiElement.inCurrentFunction(parent: PsiElement): Boolean {
     return true
 }
 
+fun PsiElement.findDeclaration(processor: PsiScopeProcessor, state: ResolveState, inCurrentScope : Boolean) : Boolean {
+    if (this is MatlabDeclaration) {
+        if (inCurrentScope || this.visibleOutsideFunction) {
+            return processor.execute(this, state)
+        }
+    }
+    return true
+}
+
 fun processDeclarations(parent: PsiElement, processor: PsiScopeProcessor, state: ResolveState, lastParent: PsiElement?, place: PsiElement): Boolean {
     val inCurrentScope = place.inCurrentFunction(parent)
 
-    val res = lastParent?.forEachSiblingBackwards { el ->
-        if (el is MatlabDeclaration) {
-            if (inCurrentScope || el.visibleOutsideFunction) {
-                return@forEachSiblingBackwards processor.execute(el, state)
-            }
+    if (!state.get(MatlabReference.IS_ORIGINAL_FILE) && lastParent != null) {
```

Comment:
I found out a problem that existed before this commit - element cannot be resolved to one of its parents.
`fib` is not resolved here:
```matlab
function res = fib(n)
    if n <= 1
      res = n;
    else
      res = fib(n-1) + fib(n-2);
    end
end
```

If I get it right you can avoid using `IS_ORIGINAL_FILE` if you solve the problem with resolve to parent.